.github/workflows/devcontainer.yml

```diff
@@ -241,3 +239,40 @@ jobs:
         docker push ${{ secrets.ACR_LOGINURL }}/devcontainer-cli:$image_tag
         docker tag mlos-devcontainer:latest ${{ secrets.ACR_LOGINURL }}/mlos-devcontainer:$image_tag
         docker push ${{ secrets.ACR_LOGINURL }}/mlos-devcontainer:$image_tag
+
+  DeployDocs:
+    if: github.ref == 'refs/heads/main'
+    needs: DevContainer
+    runs-on: ubuntu-latest
+
+    # Required for github-pages-deploy-action to push to the gh-pages branch.
+    permissions:
+      contents: write
```

Comment:
This is the key part of the change, but out of security caution, I wanted to push it to its own separate dependent job so not all other steps also have access to push to that branch of the repo.